### PR TITLE
Use Go standard library slices package more extensively

### PR DESCRIPTION
**Pull request by tklauser**
Make use of some newly added functionality added in Go 1.23.

See commit messages for details.

### CODE REVIEW

**Review by aanm — APPROVED** [review]

**Comment by aanm** [comment]
@tklauser there are some things the linter is complaining.

File: pkg/ip/ip.go [hunk 1]
```diff
@@ -755,8 +755,8 @@ func PartitionCIDR(targetCIDR net.IPNet, excludeCIDR net.IPNet) ([]*net.IPNet, [
 func KeepUniqueAddrs(addrs []netip.Addr) []netip.Addr {
 	return slices.SortedUniqueFunc(
 		addrs,
-		func(i, j int) bool {
-			return addrs[i].Compare(addrs[j]) < 0
+		func(a, b netip.Addr) int {
+			return a.Compare(b)
```
Comment (gandro) [inline pkg/ip/ip.go:759]: How does this compile? There is no change in imports here

Comment (tklauser) [reply pkg/ip/ip.go:759]: Note that `slices` here is not the Go stdlib package but `pkg/slices`, see 2nd commits which changes the signature of `SortedUniqueFunc`.

Comment (tklauser) [reply pkg/ip/ip.go:759]: ~That being said, we should probably import Go stdlib `slices` as such and alias `pkg/slices` instead.~ Scratch that, we don't import stdlib `slices` in that file.

Comment (gandro) [reply pkg/ip/ip.go:759]: Ah, I was filtering files by my codeowners, so I didn't notice the signature of `SortedUniqueFunc` changing. Thanks for the explanation!

**Review by gandro — APPROVED** [review]
Looks good for my CODEOWNERs, thanks!

**Review by dylandreimerink — APPROVED** [review]
Files owned by teams I am on seem good!

**Review by sayboras — APPROVED** [review]
Looks good for envoy related files. Thanks.

**Review by giorio94 — APPROVED** [review]
/lgtm for my codeowners

**Review by harsimran-pabla — APPROVED** [review]
BGP changes looks good. Thank you 🙏

**Review by qmonnet — APPROVED** [review]

**Review by rolinh — APPROVED** [review]
Very nice cleanup, lgtm for Hubble!

**Review by julianwiedmann — APPROVED** [review]

**Review by nathanjsweet — APPROVED** [review]

**Comment by dswaffordcw** [comment]
❤️ the simplification!  Thanks @tklauser!

**Pull request merged**